extensions/default/src/commandsModule.ts

```diff
@@ -0,0 +1,374 @@
+import { DicomMetadataStore, ServicesManager } from '@ohif/core';
+
+import DicomTagBrowser from './DicomTagBrowser/DicomTagBrowser';
+import reuseCachedLayouts from './utils/reuseCachedLayouts';
+import findViewportsByPosition, {
+  findOrCreateViewport as layoutFindOrCreate,
+} from './findViewportsByPosition';
+
+export type HangingProtocolParams = {
+  protocolId?: string;
+  stageIndex?: number;
+  activeStudyUID?: string;
+  stageId?: string;
+};
+
+const commandsModule = ({ servicesManager, commandsManager }) => {
+  const {
+    measurementService,
+    hangingProtocolService,
+    uiNotificationService,
+    viewportGridService,
+    displaySetService,
+    stateSyncService,
+    toolbarService,
+  } = (servicesManager as ServicesManager).services;
+
+  const actions = {
+    displayNotification: ({ text, title, type }) => {
+      uiNotificationService.show({
+        title: title,
+        message: text,
+        type: type,
+      });
+    },
+    clearMeasurements: () => {
+      measurementService.clear();
+    },
+
+    /**
+     * Toggles off all tools which contain a HP setter and don't match
+     * the HP id/stageId/index.
+     */
+    toggleHpTools: () => {
+      const {
+        protocol,
+        stageIndex: toggleStageIndex,
+        stage,
+      } = hangingProtocolService.getActiveProtocol();
+      const enableListener = button => {
+        if (!button.id) return;
+        const { commands, items } = button.props || button;
+        if (items) {
+          items.forEach(enableListener);
+        }
+        const hpCommand = commands?.find?.(
+          it => it?.commandName && it.commandName.indexOf('Hanging') !== -1
+        );
+        if (hpCommand) {
+          const { protocolId, stageIndex, stageId } = hpCommand.commandOptions;
+          let isActive = true;
+          if (protocolId !== undefined && protocolId !== protocol.id) {
+            isActive = false;
+          }
+          if (stageIndex !== undefined && stageIndex !== toggleStageIndex) {
+            isActive = false;
+          }
+          if (stageId && stageId !== stage.id) {
+            isActive = false;
+          }
```

Comment:
This can be refactored into

```
const isActive = (!protocolId || protocolId === protocol.id) &&
                 (!stageIndex || stageIndex === toggleStageIndex) &&
                 (!stageId || stageId === stage.id);
```